src/sentry/utils/sdk_crashes/event_stripper.py

```diff
@@ -17,77 +17,59 @@
     "contexts",
 }
 
-class EventStripper:
-    def __init__(
-        self,
-        sdk_crash_detector: SDKCrashDetector,
-    ):
-        self
-        self.sdk_crash_detector = sdk_crash_detector
-
-    ALLOWED_EVENT_KEYS = {
-        "type",
-        "datetime",
-        "timestamp",
-        "platform",
-        "sdk",
-        "level",
-        "logger",
-        "exception",
-        "debug_meta",
-        "contexts",
-    }
-
-    def strip_event_data(self, event: Event) -> Event:
-        new_event_data = dict(filter(self._filter_event, event.data.items()))
-        new_event_data["contexts"] = dict(
-            filter(self._filter_contexts, new_event_data["contexts"].items())
-        )
-
-        stripped_frames = []
-        frames = get_path(new_event_data, "exception", "values", -1, "stacktrace", "frames")
-
-        if frames is not None:
-            stripped_frames = self._strip_frames(frames)
-            new_event_data["exception"]["values"][0]["stacktrace"]["frames"] = stripped_frames
-
-        debug_meta_images = get_path(new_event_data, "debug_meta", "images")
-        if debug_meta_images is not None:
-            stripped_debug_meta_images = self._strip_debug_meta(debug_meta_images, stripped_frames)
-            new_event_data["debug_meta"]["images"] = stripped_debug_meta_images
-
-        event.data = new_event_data
-        return event
-
-    def _filter_event(self, pair):
-        key, _ = pair
-        if key in self.ALLOWED_EVENT_KEYS:
-            return True
-
-        return False
-
-    def _filter_contexts(self, pair):
-        key, _ = pair
-        if key in {"os", "device"}:
-            return True
-        return False
-
-    def _strip_debug_meta(
-        self, debug_meta_images: Sequence[Mapping[str, Any]], frames: Sequence[Mapping[str, Any]]
-    ) -> Sequence[Mapping[str, Any]]:
-
-        frame_image_addresses = {frame["image_addr"] for frame in frames}
-
-        return [
-            image for image in debug_meta_images if image["image_addr"] in frame_image_addresses
-        ]
-
-    def _strip_frames(self, frames: Sequence[Mapping[str, Any]]) -> Sequence[Mapping[str, Any]]:
-        """
-        Only keep SDK frames or non in app frames.
-        """
-        return [
-            frame
-            for frame in frames
-            if self.sdk_crash_detector.is_sdk_frame(frame) or frame.get("in_app", None) is False
-        ]
+
+def strip_event_data(event: Event, sdk_crash_detector: SDKCrashDetector) -> Event:
+    new_event_data = dict(filter(_filter_event, event.data.items()))
```

Comment:
Not sure if I already suggested this, but you could get rid of the helper method and just write:

```suggestion
    new_event_data = {key: value for key, value in event.data.items() if key in ALLOWED_EVENT_KEYS}
```